Add translation process, guidelines, working with rst files to Style Guide

catong opened the pull request:
Create files for doc translation guidelines, process, and working with rst files, and add to style guide.
Includes a sample rst file that is not included in Style Guide output - but should be accessible via a link from the style guide. (https://openedx.atlassian.net/browse/DOC-1837)
@mhoeber @srpearce @lamagnifica please take a look. This will be a WIP for a while, but at least we should be able to get something usable out.
Draft HTML is here: http://draft-translation-guide-wip.readthedocs.org/en/latest/DocTranslationProcess.html

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,58 @@
+Guidelines for Translators
+###############################################
+
+This section provides basic guidelines for translators of edx documentation.
```
edX

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,58 @@
+:ref:`Documentation Translation Process`.
+
+.. note:: Before you begin an edX translation project, you should have a basic
+   understanding of restructuredText and formatting in .rst files. For more
```
capital S in reStructuredText

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,58 @@
+
+.. note:: Before you begin an edX translation project, you should have a basic
+   understanding of restructuredText and formatting in .rst files. For more
+   information, see :ref:`Work with edX Documentation Source Files`.
```
possibly add external references to rst as well as to GitHub? http://docutils.sourceforge.net/rst.html would be my suggestion

catong replied:
Links to other rst resources are in the "Working with edX Documentation Source Files" file. This file is meant to be primarily about the translation process.

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,58 @@
+* Link text in inline links. Translate text following the tag ``:ref:``
+  Be careful not to modify the referenced link anchor.
+* Alt text in images. Translate text following the tag ``:alt:``.
+
```
examples of each of these will be helpful -- in the future

catong replied:
I have to re-evaluate/consolidate the examples and make sure I provide links.

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,58 @@
+  Files` and :ref:`SampleRSTFile<Anchor For SampleRSTFile>`. 
+* File paths in image references. 
+* The text of anchors in .rst files. 
+* Code examples that are tagged with ``..code-block::``.  
```
missing space after ..

mhoeber replied:
also don't translate inline code ``between these characters``

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,58 @@
+* Variables or database field names that are tagged with
+  monospace code font (double grave accent characters).
+
+For information about working with edX Documentation source files, see
```
lc Documnetation

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,58 @@
+  ``.. warning::`` Be careful to wrap and indent lines correctly.
+
+* Link text in inline links. Translate text following the tag ``:ref:``
+  Be careful not to modify the referenced link anchor.
```
this one is a bit tricky -- we do want them to translate text IF the link is povided after it within ``< >`` characters, otherwise not, correct? 
an interesting point is, do we want them to translate the bookmarks? I think not, since translating the header that follows should "just work".

catong replied:
Oops, thanks for the catch, I'll add the example. We definitely don't want anchors (bookmarks ) translated, and I think I make that clear elsewhere.

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,63 @@
+
+Before you begin an edX documentation translation project, you should have a
+basic understanding of restructuredText and formatting in .rst files. For more
+information, see :ref:`Work with edX Documentation Source Files`.
```
does this need to be repeated? if so, make the wording exactly the same in both places

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,63 @@
+about using Github, see https://help.git.com. You or another person on the
+translation team should understand Github repositories and the Github pull
+request process. For more information, see https://help.git.com.
+
```
repeated info ^^

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,63 @@
+#. Use Github to clone the ``edx-documentation`` repository to a location on
+   your local file system.
+
+#. Language folders, including ``en_us``, are placed one level below 
```
I suggest making a separate section on what the directory structure looks like. Then this step could have the verb closer to the beginning of the sentence, like:
#. If this is your initial translation project, duplicate the entire ``en_us`` directory in the repository at the same level, and then rename the directory to reflect your language and locale. For more information about the expected directory structure, see new section name.

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,63 @@
+
+#. Follow best practices for Github use. For information, see https://help.github.com.
+
+	* Create a working branch for your changes. Before making new branches,
```
I think they would create the branch first, and then duplicate / create the whole new language directory(?)

lamagnifica replied:
I still think that they need to create a branch, then do their directory reorganization/duplication/renaming, and then make a PR for that change. Otherwise, they're making changes directly to master, aren't they?

catong replied:
There are actually 2 "branch" actions, I guess: one to create the new language folder off master, and then another/others to create the "working branches" that I'm talking about here for actual translation work (for example, individual translators working on a branch to provide translations for particular files). Perhaps I'll pull out the initial process of creating the new language folder and say that WE own that? Then maybe once we create a new language folder on master they can start working? @mhoeber what do you think?

mhoeber replied:
working branch + "off of master."

Yes, I think we can own creating language folders

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,63 @@
+4. Build draft output and make sure everything renders as it should in both
+   HTML and PDF formats. 
+
+#. Notify the edX Documentation team when you have changes ready for review
```
how?

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,63 @@
+#. The edX Documentation team merges translation pull requests into the
+   repository.
+
+#. The edX Documentation team builds updates documentations projects for the
```
reword?

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,63 @@
+   next release or a planned future release.
+
+
+For information about working with edX Documentation source files, see
```
lc Documentation? not sure what your strategy is here -- uc the team name?

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+#################
+Sample .rst File
+#################
+
```
describe the content/purpose/audience of this file here (future)

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+
+################################
+Heading One has hash characters
+################################
```
I read most of this before I realized that this is an example of a first level heading, which is preceded and followed by # characters!
Maybe have more descriptive H1 text?

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+################################
+
+Headings are text sandwiched between rows of specific characters. The row of
+hash, asterisk, or equals characters must be as long as, or longer than, the
```
show what you mean for each of these (a lot of people say pound instead of has, for example)

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+Heading One has hash characters
+################################
+
+Headings are text sandwiched between rows of specific characters. The row of
```
row -->  rows

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+hash, asterisk, or equals characters must be as long as, or longer than, the
+text of the heading. If the translated heading string is longer than the
+original English source, make sure you extend both lines above and below the
+heading string.
```
they also have to be the same length as each other...

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+original English source, make sure you extend both lines above and below the
+heading string.
+
+This is regular text in paragraph form. There are no indents. As a best
```
maybe add a transition sentence or clause? It was hard to parse this information, since without its own heading I thought we were still talking about heading 1s...

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+line number for commenting in reviews. This is regular text in paragraph form.
+There are no indents. As a best practice, break lines at about 80 characters,
+so that each line has its own line number for commenting in reviews.
+
```
repeated info ^^

catong replied:
That was deliberate as paragraph text filler, but I don't think it's necessary to make the paragraph a certain length so I've now removed it.

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+.. comments can be added in a file by starting a line with 2 periods and a space. 
+
+In English source files, look for comments addressed to translators from writers.
+
```
.. Translators: For example, ...

catong replied:
I doubt that they should/would translate comments.

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+
+************************************
+Heading Two has asterisk characters
+************************************
```
maybe it's just me, but I'd put all the example headings together at the beginning with a description. Then, actually use the different heading to introduce discussions of the different components: notes and warnings, lists, applying formatting, links, ...

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+     your image. Do not change the value of the class attribute. For example:
+
+     **<a href="/static/Image1.jpg" class="modal-content">**
+
```
I think that should be mono rather than bold, but you probably just pasted this in from somewhere else...

catong replied:
Yep, for some reason it was bold. I changed it to mono in this example.

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+must all align, and the hyphens must all align. Empty cells must be accounted
+for, so that each column in a row is always marked, even if there is no
+content in a table cell.
+
```
I don't think this is a great example, since the 2nd column seems to always be blank. Maybe some subset from the change log?

catong replied:
I think a single example is adequate for getting the point across. A more complex example is probably nice to have but not necessary.

lamagnifica commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,214 @@
+Inline code
+===========
+
+In inline text, any text can be formatting as code (monospace font) by
```
formatting --> formatted

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+.. _Documentation Tools and Processes:
+
+*********************************
+Documentation Tools And Processes
```
lc And

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+
+For information about the tools that the edX Documentation team uses to create
+documentation, see this wiki page:
+https://openedx.atlassian.net/wiki/display/DOC/Documentation+Tools
```
is this public? we need to revisit it to see what is actually useful for translators (or move the info into another part of this document)

mhoeber replied:
I think tools pages should be public for translators and open source community

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+Lines beginning with 2 periods and a space indicate comments that are not
+visible in output. For example:
+
+``.. This is a comment.``
```
I think that the `` characters need to be removed, or explained.

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+Special Tags
+============
+
+Special tagging for notes, warnings, tables, and images is achieved using
```
I'd change images --> blocks of code since you have a whole separate section on images, and one of your examples is a code-block

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+Image Links
+============
+
+Images are included in documentation using special tagging for images and
```
"for images and providing" --> to provide

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+Each table has the number of columns and their associated relative widths
+indicated in a width tag.
+
+For proper formatting, the asterisk indicating each row must align, and the
```
align vertically?

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+
+------------------------------------------------------------
+Cross references using text of the target topic heading
+------------------------------------------------------------
```
per our Format Cheat Sheet.rstnouse file, our 3rd level headings just have a row of asterisks underneath them. 
is it possible to leverage that file more for this effort?

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+
+* H1: pound symbol (#) 
+* H2: asterisk (*)
+* H3: equals symbol (=)
```
it's a row f asterisks underneath, actually

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+where ``Exercises_introduction`` is the text for an anchor that exists
+somewhere in the files that make up the guide, and “the introduction section
+on exercises” is the link text.
+
```
watch out for curly quotes. Also, mention that curly quotes and apostrophes are super evil in this document?

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+Inline Code
+---------------
+
+In inline text, any text can be formatting as code (monospace font) by
```
formatting --> formatted

lamagnifica commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,346 @@
+*italics*.
+
+Monospace text is used for ``code examples``. Text surrounded by double back
+quotation marks (`) is rendered in monospace font. Within the double back
```
suggest being consistent about what you call these

lamagnifica commented:
This is a terrific start. I've corrected a couple of typos and made some suggesions for the future. I'm not sure how you envision the Format Cheat Sheet.rstnouse, SampleRSTFile.rst, and the other three files working together, though. I think they are pretty redundant (and occasionally contradictory) right now. That might be the only area that maybe needs to tighten up a bit before this is shared externally.

mhoeber commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,59 @@
+general guidelines.
+
+.. warning:: Do not rename or move folders or files (including .rst, .png).
+   The location and names of files in your language folder structure should
```
do we described the language folder structure?

mhoeber commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,63 @@
+	* Commit changes back to Github. 
+	* Create pull requests in Github for review purposes.
+
+4. Build draft output and make sure everything renders as it should in both
```
do we explain make html and make latexpdf anywhere?

catong replied:
I will add, or add link to a reference. Maybe future, as it will probably also involve documenting all the required config/setup files?

mhoeber commented on en_us/edx_style_guide/source/SampleRSTFile.rst:
```diff
@@ -0,0 +1,220 @@
+
+Alternative text for screen readers is required for each image. Provide text
+that is useful to someone who might not be able to see the image.
+
```
do we state somewhere that if you are providing a localized image, you must keep same file name and replace the english image?

catong replied:
Good point! I will make sure to add this.

mhoeber commented on en_us/edx_style_guide/source/WorkingWithEdXDocSource.rst:
```diff
@@ -0,0 +1,350 @@
+* What topics and concepts are covered in your course?
+* Why should a learner enroll in your course?
+
+For both numbered and bulleted lists, ensure that wrapped lines are indented
```
I'd add a note saying nested lists are supported, but you must check output because sometimes text blocks can change counts in numbered lists.

mhoeber commented:
@catong  this is great, thank you.  a few questions in line

catong commented:
Addressed review comments. Incorporated the old FormatCheatSheet into ExampleRSTFile.rst. With the most recent edits I hope it's clearer why there is one file that provides context, explanations, gotchas, etc. about working with our source files and a separate file that is intended to just provide examples of rst formatting. 
@mhoeber @lamagnifica I need to deliver the Style Guide with these updated files. Please take a look for anything egregious; (HTML version is here: http://draft-translation-guide-wip.readthedocs.org/en/latest/) 
I'm sure there's plenty to improve on in future, but for now I hope this makes sense and will provide value for the translation team.

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
`````diff
@@ -0,0 +1,104 @@
+  For example, in the following sentence, do not translate the event names that are between pairs of double grave accent characters.  ::
+
+  The edX mobile app for iOS now emits ````play_video````, ````pause_video````,
+  ````stop_video````, ````load_video````, and ````seek_video```` events. 
`````
these event names should only have one pair around each one, not two pairs: ``pause_video``

catong replied:
They needed extra pairs to show the double grave accent chars that they are described as having.

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,65 @@
+   ``fr_ca``,``fr_fr``, ``zh_tw``, ``zh_cn``, and so on. The resulting folder
+   is your working source directory.
+
+#. Follow best practices for Github use. For information, see https://help.github.com.
```
link the URL?

catong replied:
It is linked. See output.

lamagnifica commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,65 @@
+   and testing.
+
+#. Make any revisions resulting from reviews and testing, then notify the edX
+   Documentation team that the pull request is ready to merge into the repository.
```
into --> in to (?)

catong replied:
"into" is correct here.

lamagnifica commented on en_us/edx_style_guide/source/ExampleRSTFile.rst:
`````diff
@@ -0,0 +1,471 @@
+
+In inline text, any text can be formatted as code (monospace font) by
+enclosing the selection within a pair of double "grave accent" characters (`).
+For example, ````these words```` are formatted in a monospace font when the
`````
not sure about these double pairs of characters...

lamagnifica commented on en_us/edx_style_guide/source/ExampleRSTFile.rst:
```diff
@@ -0,0 +1,471 @@
+If you work with edX documentation source files, you might find this file
+helpful as a reference. This file contains examples of .rst formatting and is
+not intended to be output.
+
```
I'm a little confused by this statement paired with the extra formatting around the markup, such as :: on line 18 so that the formatting for heading levels will show in HTML output. I don't think it's a big deal, but it's a bit confusing imo

catong replied:
I'll look at this again to see if I can clarify. The sample file was not intended for output, then I wondered how translators would be able to get hold of this sample file if I didn't include it in the style guide.

lamagnifica commented:
good to go! 
:+1:

mhoeber commented on en_us/edx_style_guide/source/DocTranslationGuidelines.rst:
```diff
@@ -0,0 +1,104 @@
+
+When you translate edX documentation source .rst files, follow these general
+guidelines. For examples of each type of element, see :ref:`the Example .rst
+file<Anchor For ExampleRSTFile>`.
```
I'd add please contact docs@edx.org before starting

catong replied:
Done

mhoeber commented on en_us/edx_style_guide/source/DocTranslationProcess.rst:
```diff
@@ -0,0 +1,65 @@
+	  check that your local version of edx-documentation/master is up to date. 
+	* On your branch(es), make edits to the source files using your preferred
+	  text editor. 
+	* Save changes in the source files without renaming or moving the files.
```
maybe not now -- but we should provide build instructions to test locally and address errors as much as possible.

mhoeber commented:
@catong  this is great, couple more comments.  I'd go with this for now.  thanks

catong commented:
test this please

catong commented:
Merging https://github.com/edx/edx-documentation/pull/301
